src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/AdapterUtil.Windows.cs

```diff
@@ -29,6 +29,12 @@ internal static object LocalMachineRegistryValue(string subkey, string queryvalu
 #if !NET6_0_OR_GREATER
             (new RegistryPermission(RegistryPermissionAccess.Read, "HKEY_LOCAL_MACHINE\\" + subkey)).Assert(); // MDAC 62028
 #endif
+            // Non-Windows OS do not have registry
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return null;
+            }
+            
```

Comment:
I believe this is not necessary since the `Registry.LocalMachine.OpenSubKey` returns `null` if it can't find the key, which is fine.

Reply:
Enclosed in #if DEBUG and commented out with comment to enable only when debugging in WSL.